internal_task_pollers.go

```diff
@@ -204,6 +223,48 @@ func (wtp *workflowTaskPoller) ProcessTask(task interface{}) error {
 	return nil
 }
 
+// getNextPollRequest returns appropriate next poll request based on poller configuration.
+// Simple rules:
+// 1) if sticky execution is disabled, always poll for regular task list
+// 2) otherwise:
+//   2.1) make sure there is at least one pending request for regular task list
+//   2.2) make sure there is at least one pending request for sticky task list
+//   2.3) poll from the tasklist that has less pending requests, if they are the same, poll from sticky task list.
+// TODO: make this more smart to auto adjust based on poll latency
+func (wtp *workflowTaskPoller) getNextPollRequest() (request *s.PollForDecisionTaskRequest, release func()) {
```

Comment:
I don't understand the polling logic fully. What is the number of polling threads in this case? I think we need separate goroutine count for default and process specific task lists. Then the logic outlined in the comment wouldn't be needed as task list would be passed as an argument.

Reply:
Currently we are using 2 poller threads (goroutinues) per worker. The number of threads is controlled by the baseWorker class. The baseWorker is used by both activity worker and workflow worker. The sticky task list only apply to workflow worker. BaseWorker does not aware of the sticky task list. I'm making the poller to make the decision for each poll request on which task list to poll from. So the base worker don't need to be aware of the sticky task list thing.